test/services/PaymentsServiceSpec.scala

```diff
@@ -58,19 +58,30 @@ class PaymentsServiceSpec extends UnitSpec with MockFactory with Matchers {
 
     "the user has payments outstanding" should {
 
-      "return a list of payments" in new Test {
-        val payments = Payments(Seq(Payment(
-          "VAT Return Debit Charge",
-          LocalDate.parse("2008-12-06"),
-          LocalDate.parse("2009-01-04"),
-          LocalDate.parse("2008-12-06"),
-          BigDecimal("21.22"),
-          ""
-        )))
+      "return a list of payments sorted by due date in descending order" in new Test {
```

Comment:
For this test could you include some close boundary dates out of order to prove the sorting isn't a fluke.
2 items isn't really enough tp prove this.

e.g include payments like:
 LocalDate.parse("2008-11-06"), 
 LocalDate.parse("2008-11-05"),
 LocalDate.parse("2008-11-04"),

.. along with a couple with different months but otherwise the same, different years otherwise the same.
Then mix them up..

This will give a much better proof that the sorting is working properly 

Incidentally just as a quick test to sanity check this,
I changed the test data in this test so that the due dates were:
 LocalDate.parse("2008-11-06") and
 LocalDate.parse("2008-11-05"),

I then printed the sorted results
 sortedPayments.financialTransactions.foreach({
          println("xxxxxxxxxxxxxxxxxxx")
          payment => println(payment)}
        )

This gave the output:
xxxxxxxxxxxxxxxxxxx
Payment(VAT Return Debit Charge,2008-12-06,2009-01-04,**2008-11-05**,21.22,)
Payment(VAT Return Debit Charge,2008-12-06,2009-01-04,2008-11-06,21.22,)

This does not appear to be in descending order as indicated by test name?